src/compiler/msvc.rs

```diff
@@ -287,12 +301,58 @@ pub fn parse_arguments(arguments: &[OsString]) -> CompilerArguments<ParsedArgume
             }
         }
         match item.data {
-            Some(PreprocessorArgument) |
+            Some(PreprocessorArgument) => preprocessor_args.extend(item.arg.normalize(NormalizedDisposition::Concatenated)),
             Some(ProgramDatabase) |
             Some(DebugInfo) => common_args.extend(item.arg.normalize(NormalizedDisposition::Concatenated)),
             _ => {}
         }
     }
+
+    // TODO: doing this here reorders the arguments, hopefully that doesn't affect the meaning
```

Comment:
We'd probably have to look at the clang-cl source to figure this out, but hopefully it's not an issue. Maybe we can just make a note of it in the docs somewhere?